src/main/java/com/google/devtools/build/lib/actions/cache/MetadataCache.java

```diff
@@ -0,0 +1,12 @@
+package com.google.devtools.build.lib.actions.cache;
+
+import com.google.devtools.build.lib.actions.Artifact;
+import com.google.devtools.build.lib.actions.FileArtifactValue;
+
+public interface MetadataCache {
```

Comment:
We very likely do not need that interface -- we could just call Metadata with the metadata instead.

Reply:
Not sure whether I understand this correctly. Could you please be more specific?

Reply:
Removed `MetadataCache` interface. 

Loading metadata from action cache is happened before checking action cache (hence before action execution), we can't inject metadata with `ActionMetadataHandler` (`executionMode` is `false` at that time).

Reply:
That makes sense, thanks!